src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -580,6 +589,143 @@ public ExpressionStringPojo getConceptNormalForm(
+			@RequestParam(required = false, defaultValue = "false") boolean includeDependencies) throws ServiceException {
+		String sourceBranchPath = BranchPathUriUtil.decodePath(sourceBranch);
+		String destinationBranchPath = BranchPathUriUtil.decodePath(branch);
+
+		final List<LanguageDialect> languageDialects = List.of(new LanguageDialect(Config.DEFAULT_LANGUAGE_CODE, Long.parseLong(Concepts.US_EN_LANG_REFSET)));
+
+		// Get concepts
+		QueryService.ConceptQueryBuilder queryBuilder = queryService.createQueryBuilder(true)
+				.activeFilter(true)
+				.ecl(ecl)
+				.isReleased(true)
+				.resultLanguageDialects(languageDialects);
+
+		List<ConceptMini> conceptsFound = queryService.search(queryBuilder, sourceBranchPath, LARGE_PAGE).getContent();
+		List<ConceptMini> conceptsCreated = new ArrayList<>();
+
+		for (ConceptMini concept : conceptsFound) {
+			ConceptMini conceptCreated = copyConcept(concept, sourceBranchPath, destinationBranchPath, languageDialects, includeDependencies);
+			conceptsCreated.add(conceptCreated);
+		}
+		return conceptsCreated;
+	}
+
+	private String getModuleId(String branchPath) {
+		Branch branch = branchService.findBranchOrThrow(branchPath, true);
+		final boolean isExtension = (branch.getMetadata() != null && !Strings.isEmpty(branch.getMetadata().getString(BranchMetadataKeys.DEPENDENCY_PACKAGE)));
+
+		return isExtension ? branch.getMetadata().getString(BranchMetadataKeys.DEFAULT_MODULE_ID) : Concepts.CORE_MODULE;
+	}
+
+	private ConceptMini copyConcept(ConceptMini concept, String sourceBranchPath, String destinationBranchPath, List<LanguageDialect> languageDialects, boolean includeDependencies) throws ServiceException {
+		String sourceModuleId = getModuleId(sourceBranchPath);
+		String destinationModuleId = getModuleId(destinationBranchPath);
+
+		List<ReferenceSetMember> referenceSetMembers = new ArrayList<>();
+
+		// Get axioms
+		List<ReferenceSetMember> axiomMembers = referenceSetMemberService.findMembers(
+				sourceBranchPath,
+				new MemberSearchRequest()
+						.active(true)
+						.isNullEffectiveTime(false)
+						.referenceSet(Concepts.OWL_AXIOM_REFERENCE_SET)
+						.referencedComponentId(concept.getConceptId()), LARGE_PAGE).getContent();
+
+		// Check that the target concept exists on the destination branch
+		for (ReferenceSetMember axiomMember : axiomMembers) {
+			try {
+				SAxiomRepresentation axiomRepresentation = axiomConversionService.convertAxiomMemberToAxiomRepresentation(axiomMember);
+				if (axiomRepresentation != null) {
+					Set<Relationship> relationships = axiomRepresentation.getLeftHandSideNamedConcept() != null
+							? axiomRepresentation.getRightHandSideRelationships()
+							: axiomRepresentation.getLeftHandSideRelationships();
+
+					for (Relationship relationship : relationships) {
+						String targetConceptId = relationship.getDestinationId();
+						Concept targetConcept = conceptService.find(targetConceptId, sourceBranchPath);
```

Comment:
@astro-snail  Shouldn't the branch path be destination branch instead of source branch?

Reply:
@CoderMChu It may not exist on the destination branch, so I search on the source branch and then proceed according to the includeDependencies flag and moduleId.